src/wp-includes/post.php

```diff
@@ -2289,7 +2296,21 @@ function add_post_type_support( $post_type, $feature, ...$args ) {
 	$features = (array) $feature;
 	foreach ( $features as $feature ) {
 		if ( $args ) {
-			$_wp_post_type_features[ $post_type ][ $feature ] = $args;
+			// Check if feature already exists with args and if both are associative arrays that should be merged.
+			if (
+				isset( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
+				is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
+				isset( $args[0] ) &&
+				is_array( $args[0] )
```

Comment:
There's still no guarantee that both are associative arrays here. I think we'd need to check for `! array_is_list()` if we do this.

```suggestion
			// Check if feature already exists with args and if both are associative arrays that should be merged.
			if (
				isset( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
				is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
				( empty( $_wp_post_type_features[ $post_type ][ $feature ][0] ) || ! array_is_list( $_wp_post_type_features[ $post_type ][ $feature ][0] ) ) &&
				isset( $args[0] ) &&
				is_array( $args[0] ) &&
				! array_is_list( $args[0] )
```

Reply:
I don't think it's required for sub-params to be associative arrays. For example in future if we implemented `post-format`-esque feature it won't require an associative array.

```php
add_post_type_support( 'post', 'post-formats', array( 'aside', 'gallery' ) );
```

Reply:
OK, then the "associative" word should be removed from the comment?